test/msc3916_thumbnails_suite_test.go

```diff
@@ -75,10 +76,12 @@ func (s *MSC3916ThumbnailsSuite) TestClientThumbnails() {
 	assert.NoError(t, err)
 	assert.Equal(t, http.StatusUnauthorized, raw.StatusCode)
 
-	raw, err = client1.DoRaw("GET", fmt.Sprintf("/_matrix/client/v1/media/thumbnail/%s/%s", origin, mediaId), qs, "", nil)
-	assert.NoError(t, err)
-	assert.Equal(t, http.StatusOK, raw.StatusCode)
-	//test_internals.AssertIsTestImage(t, raw.Body) // we can't verify that the resulting image is correct
+	for _, authedClient := range []*test_internals.MatrixClient{client1, clientGuest} {
```

Comment:
generally I'd prefer to duplicate code than have errors hidden inside loop iterations. When tests fail, it's often hard to tell which client specifically caused the failure.